src/tools/perf/libperf.c

```diff
@@ -553,12 +553,10 @@ static ucs_status_t uct_perf_test_setup_endpoints(ucx_perf_context_t *perf)
         goto err_free;
     }
 
-    if (iface_attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
-        status = uct_iface_get_address(perf->uct.iface, iface_addr);
-        if (status != UCS_OK) {
-            ucs_error("Failed to uct_iface_get_address: %s", ucs_status_string(status));
-            goto err_free;
-        }
+    status = uct_iface_get_address(perf->uct.iface, iface_addr);
+    if (status != UCS_OK) {
+        ucs_error("Failed to uct_iface_get_address: %s", ucs_status_string(status));
+        goto err_free;
```

Comment:
why do you need this change? why can't you add UCT_IFACE_FLAG_CONNECT_TO_IFACE flag to your iface?